pkg/syncer/metadatasyncer.go

```diff
@@ -551,6 +573,217 @@ func InitMetadataSyncer(ctx context.Context, clusterFlavor cnstypes.CnsClusterFl
+	)
+	err := runtime.DefaultUnstructuredConverter.FromUnstructured(
+		newObj.(*unstructured.Unstructured).Object, &newNodeTopoObj)
+	if err != nil {
+		log.Errorf("topoCRUpdated: failed to cast new object %+v to %s. Error: %+v", newObj,
+			csinodetopology.CRDSingular, err)
+		return
+	}
+	err = runtime.DefaultUnstructuredConverter.FromUnstructured(
+		oldObj.(*unstructured.Unstructured).Object, &oldNodeTopoObj)
+	if err != nil {
+		log.Errorf("topoCRUpdated: failed to cast old object %+v to %s. Error: %+v", oldObj,
+			csinodetopology.CRDSingular, err)
+		return
+	}
+	// Check if there is any change in the topology labels.
+	oldTopoLabelsMap := make(map[string]string)
+	for _, label := range oldNodeTopoObj.Status.TopologyLabels {
+		oldTopoLabelsMap[label.Key] = label.Value
+	}
+	newTopoLabelsMap := make(map[string]string)
+	for _, label := range newNodeTopoObj.Status.TopologyLabels {
+		newTopoLabelsMap[label.Key] = label.Value
+	}
+	// Check if there are updates to the topology labels in the Status.
+	if reflect.DeepEqual(oldTopoLabelsMap, newTopoLabelsMap) {
+		log.Debugf("topoCRUpdated: No change in %s CR topology labels. Ignoring the event",
+			csinodetopology.CRDSingular)
+		return
+	}
+	// Ideally a CSINodeTopology CR should never be updated after the status is set to Success but
+	// in cases where this does happen, in order to maintain the correctness of domainNodeMap, we
+	// will first remove the node name from previous topology labels before adding the new values.
+	if oldNodeTopoObj.Status.Status == csinodetopologyv1alpha1.CSINodeTopologySuccess {
+		log.Warnf("topoCRUpdated: %q instance with name %q has been updated after the Status was set to "+
+			"Success. Old object - %+v. New object - %+v", csinodetopology.CRDSingular, oldNodeTopoObj.Name,
+			oldNodeTopoObj, newNodeTopoObj)
+		removeLabelsFromTopologyVCMap(ctx, oldNodeTopoObj)
+	}
+	if newNodeTopoObj.Status.Status == csinodetopologyv1alpha1.CSINodeTopologySuccess {
+		addLabelsToTopologyVCMap(ctx, newNodeTopoObj)
+	}
+}
+
+// topoCRDeleted removes the topology to VC mapping for the deleted CSINodeTopology
+// instance from the MetadataSyncer.topologyVCMap.
+func topoCRDeleted(obj interface{}) {
+	ctx, log := logger.GetNewContextWithLogger()
+	// Verify object received.
+	var nodeTopoObj csinodetopologyv1alpha1.CSINodeTopology
+	err := runtime.DefaultUnstructuredConverter.FromUnstructured(obj.(*unstructured.Unstructured).Object, &nodeTopoObj)
+	if err != nil {
+		log.Errorf("topoCRDeleted: failed to cast object %+v to %s type. Error: %+v",
+			csinodetopology.CRDSingular, err)
+		return
+	}
+	// Delete node name from domainNodeMap if the status of the CR was set to Success.
+	if nodeTopoObj.Status.Status == csinodetopologyv1alpha1.CSINodeTopologySuccess {
+		removeLabelsFromTopologyVCMap(ctx, nodeTopoObj)
+	} else {
+		log.Infof("topoCRDeleted: %q instance with name %q and status %q deleted.", csinodetopology.CRDSingular,
```

Comment:
can you add this as debug log and say you are ignoring the removal of labels as deleted CR status was not success

Reply:
Done